src/libsync/logger.cpp

```diff
@@ -325,6 +313,21 @@ void Logger::rotateLog()
                     QFile::remove(compressedName);
                 }
             }
+
+            // Expire old log files and deal with conflicts
+            {
+                auto oldLogFiles = dir.entryList(QStringList(QStringLiteral("*%1-*.log.gz").arg(qApp->applicationName())), QDir::Files, QDir::Name);
+
+                // keeping the last maxLogFiles files in total (need to subtract one from maxLogFiles to ensure the limit)
+                std::sort(oldLogFiles.begin(), oldLogFiles.end(), std::greater<QString>());
+                oldLogFiles.erase(oldLogFiles.begin(), oldLogFiles.begin() + std::min(maxLogFiles - 1, oldLogFiles.size()));
+
+                for (const auto &s : oldLogFiles) {
+                    if (!QFile::remove(dir.absoluteFilePath(s))) {
+                        qCWarning(lcUtility) << "Failed to remove old log file" << s;
```

Comment:
While we are in a runnable, calling the logger from the logger can cause issues.

Reply:
I see. I'll switch back to the original solution.